### PR TITLE
ci: add a stale bot for PRs

### PR DESCRIPTION
**Pull request by aeddi**
Closes https://github.com/gnolang/gno/issues/1445

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: .github/workflows/stale-bot.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: "Close stale PRs"
+on:
+  schedule:
+    - cron: "30 1 * * *"
+
```
Comment (ajnavarro) [inline .github/workflows/stale-bot.yml:5]: you might need these permissions for being able to add PR labels:

```
permissions:
  pull-requests: write
```

File: .github/workflows/stale-bot.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/stale@v9
+        with:
```
Comment (ajnavarro) [inline .github/workflows/stale-bot.yml:11]: do we want to check also stale issues?

File: .github/workflows/stale-bot.yml (hunk 1 above)
Comment (aeddi) [reply .github/workflows/stale-bot.yml:7]: Yes you're right!
Done.

File: .github/workflows/stale-bot.yml (hunk 2 above)
Comment (aeddi) [reply .github/workflows/stale-bot.yml:13]: I'm not sure about that so I asked the question here https://github.com/gnolang/gno/issues/1445#issuecomment-2352294634

File: .github/workflows/stale-bot.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+          stale-pr-message: "This PR is stale because it has been open 3 months with no activity. Remove stale label or comment or this will be closed in 3 months."
+          close-pr-message: "This PR was closed because it has been stalled for 3 months with no activity."
+          days-before-pr-stale: 90
+          days-before-pr-close: 90
```
Comment (moul) [inline .github/workflows/stale-bot.yml:17]:
```suggestion
          days-before-pr-close: 365
```

File: .github/workflows/stale-bot.yml
Comment (thehowl) [inline .github/workflows/stale-bot.yml:1]: Let's skip all issues and PRs with a milestone. This way we have a simple way to protect PRs and issues that have already been "planned" for a release; and we can use "Someday/maybe" to say we want stuff, but unsure when.


https://github.com/actions/stale?tab=readme-ov-file#exempt-all-milestones

File: .github/workflows/stale-bot.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
```
Comment (thehowl) [inline .github/workflows/stale-bot.yml:18]: Can we not put a warning and then close after a week at least?

Comment (thehowl) [reply .github/workflows/stale-bot.yml:18]: Nevermind, I got now what you mean.

**Comment by thehowl** [comment]
@Kouteki to be aware of the rules.

I'd say after exempting milestones, we're good to try and test this out. There may be eventually more fine-grained rules to add, or we can simply get around with the milestone exemption. Excited to see this working.

**Review by thehowl — APPROVED** [review]

File: .github/workflows/stale-bot.yml
Comment (aeddi) [reply .github/workflows/stale-bot.yml:1]: Done 👍

**Comment by Kouteki** [comment]
Relates to https://github.com/gnolang/gno/issues/3072

**Pull request merged**